sklearn/utils/testing.py

```diff
@@ -691,14 +693,28 @@ def run_test(*args, **kwargs):
 
 def skip_if_32bit(func):
     """Test decorator that skips tests on 32bit platforms."""
-    @wraps(func)
-    def run_test(*args, **kwargs):
-        bits = 8 * struct.calcsize("P")
-        if bits == 32:
-            raise SkipTest('Test skipped on 32bit platforms.')
-        else:
-            return func(*args, **kwargs)
-    return run_test
+    if inspect.isgeneratorfunction(func):
```

Comment:
I changed it so if skip_if_32bit is called with a generator it recalls skip_if_32bit with the functions it generates. This maintains the original way the function was written and removes duplicate code. It does introduce technically introduce a recursive call, but in practice the recursion depth should never exceed 1.